Add instructor LTI 1.3 documentation (#10863)
PrairieLearn/PrairieLearn: docs/lti13Instructors.md

# LTI 1.3 configuration for instructors

> LTI 1.3 requires a trust relationship between the Learning Management System (Canvas) and

Reviewer: I know the current beta is only concerned about Canvas, but I'd imagine this will eventually work for other LMSs as well (Moodle, etc.), right? Would it make more sense to list Canvas as a restriction of the beta (or public preview as mentioned by @mwest1066 below) instead of referring to Canvas as the only LMS?

PrairieLearn. We will need to set that up with your LMS admin before LTI is available at
the course level.

LTI 1.3 is available in beta. Reach out to support@prairielearn.com to get it set up.

## Setting up your Canvas course for PrairieLearn

In Canvas, if PrairieLearn LTI 1.3 is available for your course, it will be in the list of
items under Settings / Navigation. It's in the hidden list by default. Drag PrairieLearn
to the top visible list and click Save. When the page reloads, you should see PrairieLearn
in your Canvas course left menu.

The next step is to setup a connection between the Canvas course and a PrairieLearn course
instance. As a Canvas Teacher, click the PrairieLearn left menu link. You will be presented
a page to select a course instance to link.

If students click the left menu PrairieLearn link before you have connected a course instance,
they will see a 'come back later' message. After a course instance is connected, the link will
take them into your course instance.

> ## LTI 1.3 in the course instance

Reviewer 2: ```suggestion
## "LTI 1.3" tab in the PrairieLearn course instance
```

> Once a Canvas course is linked with your course instance, you will see an LTI 1.3 tab
> in the course instance instructor view. All of the LTI 1.3 features are managed from
> that page.

Reviewer 2: ```suggestion
Once a Canvas course is linked with your PrairieLearn course instance, you will see an "LTI 1.3" tab
in the PrairieLearn course instance instructor view. All of the Canvas integration features are managed from
that page.
```

> ### Linking assessments and sending grades
> 
> To be able to push grades from PrairieLearn to Canvas, you need to link the PrairieLearn
> assessment with a Canvas assignment. You can either use PrairieLearn to create a new
> assignment in Canvas based off the PrairieLearn assessment, or you can poll Canvas for
> assignments available to PrairieLearn and pick from pre-existing assignments.
> 
> PrairieLearn can only see assignments in Canvas that are associated with PrairieLearn.
> To do this in Canvas, edit the assignment Submission Type to "External Tool" and set
> the External Tool URL to the students' link to the assessment. (You can get this URL
> from the assessment instructor settings page.)
> 
> When a PrairieLearn assessment is linked with a Canvas assignment, you will see a
> "Send Grades" button. Use this button to manually push grades to Canvas.

Reviewer 2: Please check this for accuracy! In particular the bit about how grades are sent as percentages.
```suggestion
## Linking assessments

To be able to send grades from PrairieLearn to Canvas, you first need to link the PrairieLearn
assessment with a Canvas assignment. There are two ways to do this:

1. **Create a new Canvas assignment for the PrairieLearn assessment.** In PrairieLearn, go to your course instance and then the "LTI 1.3" tab. Click the "Link assignment" button and then the "Create a new assignment" button. This will create a new Canvas assignment and link it to the PrairieLearn assessment.
2. **Link an existing Canvas assignment.** If you already have a Canvas assignment that you want to link to a PrairieLearn assessment, there is a multi-step procedure:
    1. In PrairieLearn, go to the assessment, then the "Settings" tab, and copy the "Student Link".
    2. In Canvas, go to the Assignments page. Next to your assignment click the three-dots menu and select "Edit". Change the "Submission Type" to "External Tool" and set the "External Tool URL" to the student link that you copied above from PrairieLearn.
    3. In PrairieLearn, go to your course instance and then the "LTI 1.3" tab. Click the "Link assignment" button and then the "Pick from existing" button. Select the Canvas assignment and click "Link assignment".

## Sending grades from PrairieLearn to Canvas

To send grades you first need to link the PrairieLearn assessment with a Canvas assignment, as described above. After doing this, go to your PrairieLearn assessment, select the "LTI 1.3" tab, and then click "Send grades". Grades are always sent as percentage scores out of 100. These are the percentage "Score" values that PrairieLearn shows on the "Gradebook" page and the "Students" tab inside each assessment.

## Unlinking assessments

You can unlink a single PrairieLearn assessment from Canvas. This can be helpful if you want to link it to a different Canvas assignment. Unlinking does not change any grades in Canvas or PrairieLearn.

To do this, go to your course instance in PrairieLearn, then the "LTI 1.3" tab. For the assessment you want to unlink, select the dropdown arrow on the right of the "Send grades" button and select "Unlink assignment".
```

> ### Connection to LMS

Reviewer 2: ```suggestion
## Removing the connection to Canvas
```

> You can remove PrairieLearn's connection with that Canvas course from the LTI 1.3
> instructor page. This is useful if you want to start over or link the Canvas course
> with a different PrairieLearn course instance.

Reviewer 2: ```suggestion
You can remove PrairieLearn's connection with Canvas for an entire course instance. This is useful if you want to start over or link the Canvas course
with a different PrairieLearn course instance. Removing this connection does not change any grades in either Canvas or PrairieLearn.

To do this, go to your course instance in PrairieLearn, select the "LTI 1.3" tab, and click the "Remove LTI 1.3 connection" at the bottom of the page.
```

> 
> # LTI 1.3 future planning
> 
> LTI 1.3 is under active development and beta testing.
> 
> Our only currently supported Learning Management System is Canvas. Supporting
> additional learning systems will be explored after Canvas support comes out of beta.
> 
> Sending grades is currently a manual operation for instructors. We will add the
> configurable ability for PrairieLearn to push grades continuously so Canvas is
> always in sync with PrairieLearn grades.

Reviewer 2: ```suggestion
```
